process/factory/shard/preProcessorsContainerFactory.go

```diff
@@ -84,6 +88,9 @@ func NewPreProcessorsContainerFactory(
 	if rewardsProducer == nil || rewardsProducer.IsInterfaceNil() {
 		return nil, process.ErrNilInternalTransactionProducer
 	}
+	if economicsFee == nil {
```

Comment:
Maybe you can also create IsInterfaceNil method and check in the same way as the other interfaces are checked here for nil

Reply:
Done.